### PR TITLE
Update discussion and social media links.

### PR DESCRIPTION
**Pull request by kevinbowrin**
Per the call out at https://fosstodon.org/@beeware/110053388972910487, this PR updates the project's discussion forum and social media links in the documentation. 

I also did a tiny amount of cleanup and reorganizing, mostly moving the 'footer' style links to just after their sections. That made it easier to ensure consistency between the different README docs.

## PR Checklist:

- [x] All new features have been tested
- [x] All new features have been documented
- [x] I have read the **CONTRIBUTING.md** file
- [x] I will abide by the code of conduct

### CODE REVIEW

File: changes/1822.misc.rst [hunk 1]
```diff
@@ -0,0 +1,7 @@
+Docs - update discussion and social media links.
```
Comment (freakboy3742) [inline changes/1822.misc.rst:1]: The changenote in this file should be a 1 line description, suitable for inclusion in release notes. It generally shouldn't discuss specifics of the change. 
```suggestion
Social media and discussion links were updated.
```

**Review by freakboy3742 — CHANGES_REQUESTED** [review]
Extremely thorough update - thanks for the PR!

I've flagged one change inline, with a suggested fix. Ordinarily, I'd be happy to make that change myself and land the PR; but you've made this PR on your main branch, which I can't edit by default. As a rule, it's generally a good idea to submit PRs from a specific branch, rather than the main branch; this will also make it a lot easier to keep your repo in sync with the upstream repo.

**Comment by kevinbowrin** [comment]
The `changes/1822.misc.rst` file is now simply:

```
Social media and discussion links were updated.
```

**Review by freakboy3742 — APPROVED** [review]
Awesome - thanks!

**Pull request merged**